core/src/main/java/com/datastax/oss/driver/internal/core/type/codec/extras/array/BooleanArrayCodec.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright DataStax, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.datastax.oss.driver.internal.core.type.codec.extras.array;
+
+import com.datastax.oss.driver.api.core.ProtocolVersion;
+import com.datastax.oss.driver.api.core.type.DataTypes;
+import com.datastax.oss.driver.api.core.type.reflect.GenericType;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.nio.ByteBuffer;
+import java.util.Objects;
+import net.jcip.annotations.Immutable;
+
+/**
+ * A codec that maps the CQL type {@code list<boolean>} to the Java type {@code boolean[]}.
+ *
+ * <p>Note that this codec is designed for performance and converts CQL lists <em>directly</em> to
+ * {@code boolean[]}, thus avoiding any unnecessary boxing and unboxing of Java primitive {@code
+ * boolean} values; it also instantiates arrays without the need for an intermediary Java {@code
+ * List} object.
```

Comment:
Those comments would be more useful on the constants and methods in `TypeCodecs`.

The codec implementations themselves are internal, the end user does not usually reference them directly.